### ISSUE TITLE
Header, Side Menu, and Footer: Add landmarks to examples

### ISSUE THREAD

**Issue by bdfranck** [GovAlta/ui-components#3136]
Update the examples in our Header, Footer, and Side Menu documentation to include with their respective [HTML landmarks](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/landmark_role).

## Acceptance Criteria

- Header code examples are wrapped in `<header>`
- Side menu code examples are wrapped in `<nav>`
- Footer code examples are wrapped in `<footer>`

## Context

From the [original ticket](https://github.com/GovAlta/ui-components/issues/2362):

> This was raised in the UAIM Accessibility Reported as a Level 3 Issue. The issue is as reported, "Regions (footer, etc.) not labelled correctly; could easily get lost and not really have a mental map of the site". The recommendation is to use proper descriptive landmarks to help users understand the structure and navigate more effectively.

## Example

```html
<footer>
  <GoAAppFooter maxContentWidth="100%">
    <GoAAppFooterMetaSection>
      <a href="#">
        Give feedback
      </a>
      <a href="#">
        Accessibility
      </a>
      <a href="#">
        Privacy
      </a>
      <a href="#">
        Contact us
      </a>
    </GoAAppFooterMetaSection>
  </GoAAppFooter>
</footer>
```
